content/articles/develop-and-deploy-fast-apps-with-vite-js/index.md

```diff
@@ -0,0 +1,152 @@
+
+The instructions include:
+
+![Vanilla.js](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/vanilla-js.jpg)
+
+We will cd into our project and run the npm install. This means changing the directory of our project and installing all the dependencies.
+
+![Instructions](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/instructions.jpg)
+
+Before running the `vite-project`, this is its structure after installing dependencies. We have a front-end structure with some `node modules` for our dependencies. We have a default `index.html` file. The CSS file gets imported at the `main.js` file.
+
+![Installed dependencies](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/installed-dependencies.jpg)
+
+The project structure is pretty simple. Let us look at what the `package.json` file contains. It has three scripts that we're going to use:
+- dev script - start our local folder.
+- build script - bundle our project for production.
+- preview script - look at our local build in our local environment.
+
+![Package.json](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/package-json.jpg)
+
+Since we have installed the dependencies, let's run `npm run dev`. Then, open the project at [localhost 3000](http://localhost:3000/). Click the link, and it will open on your browser.
+
+Congratulations! You now have Vanilla.js and Vite running on your local server. You will notice the `hmr update` running when you edit files and changes appear instantly on the browser.
+
+![Launch the project on the browser](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/npm-run.jpg)
+
+![Vite project](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/vite-project.jpg)
+
+To begin building or developing the project, run `npm run build`. The project is compiled into a `dist` folder containing the JavaScript and CSS files when that command is executed. Both files appear to be hashed.
+
+![Dist](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/dist.jpg)
+
+We execute the command `npm run preview` to preview the project. It will serve as a preview on [localhost 4173](http://localhost:4173/). So, 3000 is your build server, and 4173 is the preview for our build.
+
+![Preview](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/preview.jpg)
+
+#### Build a JavaScrip app to learn more about Vite.
+Let us create a simple JavaScript application to learn more about Vite. We will create a file in our project's folder called `index.md`. The markdown file will have some data for us to load.
+
+File `index.md`:
+
+![Markdown file](/engineering-education/develop-and-deploy-fast-apps-with-vite-js/markdown-cop.jpg)
+
+One of the awesome features of Vite is that you can easily import data into your JavaScript. We will need to install markdownIt `npm install markdown-it`. This will let us process text into markdown directly in my application.
```

Comment:
```suggestion
One of the awesome features of Vite is that you can easily import data into your JavaScript application. We will need to install markdownIt using the command  `npm install markdown-it`. This will let us process text into markdown directly in my application.